diff --git a/dist/routers/createRouter.js b/dist/routers/createRouter.js
index c6216015d92280f70d3872064a214159142c8542..25094e8ed725144558a2f0d579fe45fda3f96c6a 100644
--- a/dist/routers/createRouter.js
+++ b/dist/routers/createRouter.js
@@ -43,7 +43,7 @@ export function bindEvent(target, type, handler) {
     return () => target.removeEventListener(type, handler);
 }
 export function scrollToHash(hash, fallbackTop) {
-    const el = querySelector(`#${hash}`);
+    const el = querySelector(`#${hash}`) ?? document.getElementById(hash);
     if (el) {
         el.scrollIntoView();
     }
