diff --git a/worker.js b/worker.js
index 8a8a802ca719a55f4d2ae526eb54e712edfd7455..dc705b3b8421e2ea3fb664029f0715757dd0dda7 100644
--- a/worker.js
+++ b/worker.js
@@ -43,7 +43,7 @@ runAsWorker(
       : null;
 
     const { ignored, inferredParser } = await prettier.getFileInfo(
-      onDiskFilepath,
+      filepath,
       {
         resolveConfig: false,
         withNodeModules: false,
@@ -161,7 +161,7 @@ runAsWorker(
         'svelte',
         'pug',
       ];
-      if (parserBlocklist.includes(/** @type {string} */ (inferredParser))) {
+      if (false && parserBlocklist.includes(/** @type {string} */ (inferredParser))) {
         return;
       }
     }
