diff --git a/lib/index.js b/lib/index.js
index 3c6e342d9933d8fde32018334dcd64ab436606c5..c421c514b509cba173d3c74f762cb5e29bcac219 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -741,7 +741,8 @@ function containerFlow(parent, state, info) {
 function inferDepth(state) {
   let depth = 0
 
-  for (const x of state.stack) {
+  for (const x of [...state.stack].reverse()) {
+    if (x === 'listItem') break;
     if (x === 'mdxJsxFlowElement') {
       depth++
     }
