diff --git a/index.js b/index.js
index e6456c91f9ebe098deb4c73dbb544c156c2d17e7..1284ed3cc4551251079b71093dedb513d9db50dd 100644
--- a/index.js
+++ b/index.js
@@ -206,6 +206,7 @@
 import {lintRule} from 'unified-lint-rule'
 import {visit, SKIP} from 'unist-util-visit'
 import {pointStart, pointEnd} from 'unist-util-position'
+import stringWidth from 'string-width'
 
 const remarkLintTableCellPadding = lintRule(
   {
@@ -226,6 +227,8 @@ const remarkLintTableCellPadding = lintRule(
       )
     }
 
+    const value = String(file)
+
     visit(tree, 'table', (node) => {
       const rows = node.children
       // To do: fix types to always have `align` defined.
@@ -285,7 +288,7 @@ const remarkLintTableCellPadding = lintRule(
           sizes[column] = Math.max(
             /* c8 ignore next */
             sizes[column] || 0,
-            contentEnd - contentStart
+            stringWidth(value.substring(contentStart, contentEnd))
           )
         }
       }
@@ -328,7 +331,7 @@ const remarkLintTableCellPadding = lintRule(
 
       if (style === 0) {
         // Ignore every cell except the biggest in the column.
-        if (size(cell) < sizes[column]) {
+        if (size(cell, value) < sizes[column]) {
           return
         }
 
@@ -338,7 +341,7 @@ const remarkLintTableCellPadding = lintRule(
 
         if (spacing > style) {
           // May be right or center aligned.
-          if (size(cell) < sizes[column]) {
+          if (size(cell, value) < sizes[column]) {
             return
           }
 
@@ -360,12 +363,14 @@ export default remarkLintTableCellPadding
 
 /**
  * @param {TableCell} node
+ * @param {string} value
  * @returns {number}
  */
-function size(node) {
+function size(node, value) {
   const head = pointStart(node.children[0]).offset
   const tail = pointEnd(node.children[node.children.length - 1]).offset
   // Only called when we’re sure offsets exist.
-  /* c8 ignore next */
-  return typeof head === 'number' && typeof tail === 'number' ? tail - head : 0
+  if (typeof head !== 'number' || typeof tail !== 'number') return 0
+  const content = value.substring(head, tail)
+  return stringWidth(content)
 }
diff --git a/package.json b/package.json
index aa1c8caac219f9c7ba22a4c983cb9f8331bd6bbd..a722d5599f55d4817c9dc19a1d84672c47902d72 100644
--- a/package.json
+++ b/package.json
@@ -37,6 +37,7 @@
   "dependencies": {
     "@types/mdast": "^3.0.0",
     "@types/unist": "^2.0.0",
+    "string-width": "^5",
     "unified": "^10.0.0",
     "unified-lint-rule": "^2.0.0",
     "unist-util-position": "^4.0.0",
