diff --git a/index.js b/index.js
index 516056c8130427d6e85b335c94ffb18fcd8230c2..720cff21a0c28abfcb70f3fd2304ac31dcc4df71 100644
--- a/index.js
+++ b/index.js
@@ -439,6 +439,7 @@ import {lintRule} from 'unified-lint-rule'
 import {pointEnd, pointStart} from 'unist-util-position'
 import {SKIP, visitParents} from 'unist-util-visit-parents'
 import {VFileMessage} from 'vfile-message'
+import stringWidth from 'string-width'
 
 const remarkLintTableCellPadding = lintRule(
   {
@@ -467,6 +468,7 @@ const remarkLintTableCellPadding = lintRule(
      * @property {number} middle
      * @property {number | undefined} right
      * @property {Point} rightPoint
+     * @property {number} visual
      */
 
     const value = String(file)
@@ -516,9 +518,9 @@ const remarkLintTableCellPadding = lintRule(
         if (
           entry.size &&
           (sizes[entry.column] === undefined ||
-            entry.size.middle > sizes[entry.column])
+            entry.size.visual > sizes[entry.column])
         ) {
-          sizes[entry.column] = entry.size.middle
+          sizes[entry.column] = entry.size.visual
         }
       }
 
@@ -527,8 +529,8 @@ const remarkLintTableCellPadding = lintRule(
         for (const info of entries) {
           if (
             info.size &&
-            info.size.middle &&
-            info.size.middle === sizes[info.column]
+            info.size.visual &&
+            info.size.visual === sizes[info.column]
           ) {
             const node = info.ancestors.at(-1)
             assert(node) // Always defined.
@@ -581,7 +583,7 @@ const remarkLintTableCellPadding = lintRule(
         return
       }
 
-      const alignSpaces = sizes[info.column] - info.size.middle
+      const alignSpaces = sizes[info.column] - info.size.visual
       const min = expected === 'compact' ? 0 : 1
       /** @type {number} */
       let max = min
@@ -594,7 +596,7 @@ const remarkLintTableCellPadding = lintRule(
 
       // For empty cells,
       // the `left` field is used for all the whitespace in them.
-      if (info.size.middle === 0) {
+      if (info.size.visual === 0) {
         if (side === 'right') return
         max = Math.max(max, sizes[info.column] + 2 * min)
       }
@@ -881,7 +883,8 @@ const remarkLintTableCellPadding = lintRule(
             line: end.line,
             column: end.column - (end.offset - rightEdgeIndex),
             offset: rightEdgeIndex
-          }
+          },
+          visual: stringWidth(value.slice(leftIndex, rightIndex))
         }
       }
     }
diff --git a/package.json b/package.json
index a7cae652e84349da5ee20768fc3aeb4fe8447b2f..edb1f0bc519a6ec86fdf4f4c15fae300d7648e6c 100644
--- a/package.json
+++ b/package.json
@@ -10,6 +10,7 @@
     "devlop": "^1.0.0",
     "mdast-util-phrasing": "^4.0.0",
     "pluralize": "^8.0.0",
+    "string-width": "^7",
     "unified-lint-rule": "^3.0.0",
     "unist-util-position": "^5.0.0",
     "unist-util-visit-parents": "^6.0.0",
