diff --git a/dist/index.js b/dist/index.js
index b4fac7103ac3747ffe6c524912105c5448add2ee..7300f7129b2fe9944a882212db8fd6b041004eba 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -49,6 +49,12 @@ function imagetools(userOptions = {}) {
     return {
         name: 'imagetools',
         enforce: 'pre',
+        filter: {
+          id: {
+            include: pluginOptions.include,
+            exclude: pluginOptions.exclude,
+          }, 
+        },
         configResolved(cfg) {
             viteConfig = cfg;
             basePath = createBasePath(viteConfig.base);
@@ -65,7 +71,7 @@ function imagetools(userOptions = {}) {
             const lazyLoadImage = () => {
                 if (lazyImg)
                     return lazyImg;
-                return (lazyImg = sharp(pathname));
+                return (lazyImg = sharp(pathname, { animated: true }));
             };
             let lazyMetadata;
             const lazyLoadMetadata = async () => {
@@ -112,7 +118,7 @@ function imagetools(userOptions = {}) {
                 let cachedBuffer;
                 if (cacheOptions.enabled && ((_b = (_a = statSync(`${cacheOptions.dir}/${id}`, { throwIfNoEntry: false })) === null || _a === void 0 ? void 0 : _a.size) !== null && _b !== void 0 ? _b : 0) > 0) {
                     cachedBuffer = await readFile(`${cacheOptions.dir}/${id}`);
-                    image = sharp(cachedBuffer);
+                    image = sharp(cachedBuffer, { animated: true });
                     metadata = (await image.metadata());
                     // we set the format on the metadata during transformation using the format directive
                     // when restoring from the cache, we use sharp to read it from the image and that results in a different value for avif images
