diff --git a/dist/index.js b/dist/index.js
index 4fbba19..134d985 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -67,7 +67,7 @@ function imagetools(userOptions = {}) {
                 const lazyLoadImage = () => {
                     if (lazyImg)
                         return lazyImg;
-                    return (lazyImg = sharp(pathname));
+                    return (lazyImg = sharp(pathname, { animated: true }));
                 };
                 let lazyMetadata;
                 const lazyLoadMetadata = async () => {
@@ -115,7 +115,7 @@ function imagetools(userOptions = {}) {
                     if (cacheOptions.enabled &&
                         ((_b = (_a = statSync(`${cacheOptions.dir}/${id}`, { throwIfNoEntry: false })) === null || _a === void 0 ? void 0 : _a.size) !== null && _b !== void 0 ? _b : 0) > 0) {
                         cachedBuffer = await readFile(`${cacheOptions.dir}/${id}`);
-                        image = sharp(cachedBuffer);
+                        image = sharp(cachedBuffer, { animated: true });
                         metadata = (await image.metadata());
                         // we set the format on the metadata during transformation using the format directive
                         // when restoring from the cache, we use sharp to read it from the image and that results in a different value for avif images
