diff --git a/client.d.ts b/client.d.ts
index 3a9dd88f771f0d88310d8441543af477269e3973..d87d395976eb895a13b592bd10fed379d1f9caaf 100644
--- a/client.d.ts
+++ b/client.d.ts
@@ -58,18 +58,6 @@ declare module '*.bmp' {
   const src: string
   export default src
 }
-declare module '*.png' {
-  const src: string
-  export default src
-}
-declare module '*.jpg' {
-  const src: string
-  export default src
-}
-declare module '*.jpeg' {
-  const src: string
-  export default src
-}
 declare module '*.jfif' {
   const src: string
   export default src
@@ -82,26 +70,10 @@ declare module '*.pjp' {
   const src: string
   export default src
 }
-declare module '*.gif' {
-  const src: string
-  export default src
-}
-declare module '*.svg' {
-  const src: string
-  export default src
-}
 declare module '*.ico' {
   const src: string
   export default src
 }
-declare module '*.webp' {
-  const src: string
-  export default src
-}
-declare module '*.avif' {
-  const src: string
-  export default src
-}
 declare module '*.cur' {
   const src: string
   export default src
